onnxruntime/python/onnxruntime_pybind_state.cc

```diff
@@ -1570,33 +1570,21 @@ void InitArray() {
   })();
 }
 
-// static variable used to create inference session and training session.
-static std::unique_ptr<Environment> session_env;
-
-void InitializeEnv() {
-  auto initialize = [&]() {
-    // Initialization of the module
-    InitArray();
-    Env::Default().GetTelemetryProvider().SetLanguageProjection(OrtLanguageProjection::ORT_PROJECTION_PYTHON);
-    OrtPybindThrowIfError(Environment::Create(std::make_unique<LoggingManager>(
-                                                  std::make_unique<CLogSink>(),
-                                                  Severity::kWARNING, false, LoggingManager::InstanceType::Default,
-                                                  &SessionObjectInitializer::default_logger_id),
-                                              session_env));
-
-    static bool initialized = false;
-    if (initialized) {
-      return;
-    }
-    initialized = true;
-  };
-  initialize();
+static std::unique_ptr<Environment> InitializeEnv() {
+  // Initialization of the module
+  std::unique_ptr<Environment> ret;
+  InitArray();
+  Env::Default().GetTelemetryProvider().SetLanguageProjection(OrtLanguageProjection::ORT_PROJECTION_PYTHON);
+  OrtPybindThrowIfError(Environment::Create(std::make_unique<LoggingManager>(
+                                                std::make_unique<CLogSink>(),
+                                                Severity::kWARNING, false, LoggingManager::InstanceType::Default,
+                                                &SessionObjectInitializer::default_logger_id),
+                                            ret));
+  return ret;
 }
 
 onnxruntime::Environment& GetEnv() {
-  if (!session_env) {
-    InitializeEnv();
-  }
+  static std::unique_ptr<Environment> session_env = InitializeEnv();
```

Comment:
It deletes it earlier because function statics are deleted before global statics?

Reply:
Yes